lib/recursive-methods.rb

```diff
@@ -1,49 +1,71 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
+# https://www.rubyguides.com/2015/08/ruby-recursion-and-memoization/
 
-# Time complexity: ?
-# Space complexity: ?
-def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+# Time complexity: O(1)
+# Space complexity: O(n)
+
+def factorial(m)
+  if m == 0 || m == 1
+    return 1 #bc it needs to return 1
+  elsif m < 0
+    raise ArgumentError, "sorrrry"
 end
 
-# Time complexity: ?
-# Space complexity: ?
+  return m * factorial(m-1)
+end
+
+# Time complexity: O(n)
+# Space complexity: O(n)
+#number of stacks is what were gaging
+# how many times are we visiting each char?
+# time complexity isnt in numbers, it depends on input
+# how does it scale?
 def reverse(s)
-    raise NotImplementedError, "Method not implemented"
+  if s.length == 1 #basecase
+    return s 
+  else
+    return reverse(s[1..-1]) + s[0]
```

Comment:
Missing an end to this if statement.

```suggestion
    return reverse(s[1..-1]) + s[0]
  end
```